### PR TITLE
Silence some warnings

### PR DESCRIPTION
**Pull request by Megamouse**

### CODE REVIEW

File: rpcs3/Emu/RSX/RSXThread.cpp [hunk 1]
```diff
@@ -50,7 +50,7 @@ bool serialize<rsx::frame_capture_data>(utils::serial& ar, rsx::frame_capture_da
 {
 	ar(o.magic, o.version, o.LE_format);
 
-	if (o.magic != rsx::c_fc_magic || o.version != rsx::c_fc_version || o.LE_format != (std::endian::little == std::endian::native))
+	if (o.magic != rsx::c_fc_magic || o.version != rsx::c_fc_version || o.LE_format != +(std::endian::little == std::endian::native))
```
Comment (elad335) [inline rpcs3/Emu/RSX/RSXThread.cpp:53]:
```suggestion
	if (o.magic != rsx::c_fc_magic || o.version != rsx::c_fc_version || o.LE_format != u32{std::endian::little == std::endian::native})
```

File: rpcs3/Emu/Cell/SPUThread.cpp [hunk 2]
```diff
@@ -3205,7 +3205,7 @@ bool spu_thread::process_mfc_cmd()
 			{
 				state += cpu_flag::wait + cpu_flag::temp;
 				std::this_thread::yield();
-				!check_state();
+				[[maybe_unused]] bool res = check_state();
```
Comment (MSuih) [inline rpcs3/Emu/Cell/SPUThread.cpp:3208]: static cast to void instead, it's far less wordy than this.

File: rpcs3/Emu/Cell/lv2/sys_net.cpp [hunk 3]
```diff
@@ -2320,8 +2320,8 @@ error_code sys_net_bnet_recvfrom(ppu_thread& ppu, s32 s, vm::ptr<void> buf, u32
 			{
 				const auto get_data = [&](unsigned char *dest_buf)
 				{
-					const u32 to_give = std::min(sock.p2ps.data_available, len);
-					sys_net.trace("STREAM-P2P socket had %d available, given %d", sock.p2ps.data_available, to_give);
+					const u32 to_give = std::min<u32>(sock.p2ps.data_available, len);
+					sys_net.trace("STREAM-P2P socket had %lu available, given %d", sock.p2ps.data_available, to_give);
```
Comment (elad335) [inline rpcs3/Emu/Cell/lv2/sys_net.cpp:2324]:
```suggestion
					sys_net.trace("STREAM-P2P socket had %u available, given %u", sock.p2ps.data_available, to_give);
```

Comment (Megamouse) [reply rpcs3/Emu/Cell/lv2/sys_net.cpp:2324]: why? it's u64

Comment (elad335) [reply rpcs3/Emu/Cell/lv2/sys_net.cpp:2324]: It's fmt::format, not prinf.

File: rpcs3/Emu/Cell/lv2/sys_net.cpp [hunk 4]
```diff
@@ -3273,7 +3273,7 @@ error_code sys_net_bnet_poll(ppu_thread& ppu, vm::ptr<sys_net_pollfd> fds, s32 n
 					{
 						if ((fds[i].events & SYS_NET_POLLIN) && sock->p2ps.data_available)
 						{
-							sys_net.trace("[P2PS] p2ps has %d bytes available", sock->p2ps.data_available);
+							sys_net.trace("[P2PS] p2ps has %lu bytes available", sock->p2ps.data_available);
```
Comment (Megamouse) [inline rpcs3/Emu/Cell/lv2/sys_net.cpp:3276]:
```suggestion
							sys_net.trace("[P2PS] p2ps has %u bytes available", sock->p2ps.data_available);
```

File: rpcs3/Emu/Cell/SPUThread.cpp (hunk 2 above)
Comment (Megamouse) [inline rpcs3/Emu/Cell/SPUThread.cpp:3208]:
```suggestion
				static_cast<void>(check_state());
```

**Pull request merged**